test/modes_internal_test.c

```diff
@@ -197,76 +188,65 @@ static int execute_cts128(CTS128_FIXTURE fixture)
     unsigned char cleartext[64], ciphertext[64], vector[64];
     size_t tail;
 
-    TEST_info("%s_vector_%lu", fixture.case_name, (unsigned long)len);
+    TEST_info("%s_vector_%lu", fixture->case_name, (unsigned long)len);
 
-    tail = fixture.transform_output(orig_vector, vector, len);
+    tail = fixture->transform_output(orig_vector, vector, len);
 
     /* test block-based encryption */
     memcpy(iv, test_iv, test_iv_len);
-    fixture.encrypt_block(test_input, ciphertext, len,
-                          encrypt_key_schedule, iv,
-                          (block128_f)AES_encrypt);
-    if (!TEST_mem_eq(ciphertext, len, vector, len))
-        return 0;
-
-    if (!TEST_mem_eq(iv, sizeof(iv), vector + len - tail, sizeof(iv)))
+    fixture->encrypt_block(test_input, ciphertext, len, encrypt_key_schedule,
+                           iv, (block128_f)AES_encrypt);
```

Comment:
On side note encrypt_block returns a value, it would be appropriate to check it [for equality to len].